### DISCUSSION TITLE
Rename "DNI estimation" and "Decomposing & combining irrad" sections for clarity

### DISCUSSION THREAD

**Discussion by mikofski** [pvlib/pvlib-python#1696] [Ideas]
As discussed in https://github.com/pvlib/pvlib-python/pull/1179#discussion_r1131064988 and https://github.com/pvlib/pvlib-python/pull/1179#discussion_r1131047092 the decomposition functions differ - while Erbs and Boland yield the diffuse fraction (DF) DISC and DIRINT return DNI. Therefore, the documentation subheading "[DNI estimation](https://pvlib-python.readthedocs.io/en/stable/reference/irradiance/decomposition.html)" can be misleading if not all of the functions directly return DNI. Also, these methods (DISC, DIRINT, Erbs, Boland) are often called "Decomposition" methods. However, there is already a subsection called "[Decomposing & Combining Irradiance](https://pvlib-python.readthedocs.io/en/stable/reference/irradiance/components.html)" which contains methods for deriving DNI from GHI & DHI, deriving AOI, calculating extra-terrestrial irradiance, and getting the ground diffuse.

Do folks think these sections subheading could be improved? What would be an improvement? In particular, who would be in favor of switching and renaming these sections like:
* "DNI estimation models" --> "Decomposition models"?
* "Decomposing and combining irradiance" --> "Combing & Calculating Irradiance Components"? or something like that?

**Comment by cwhanse** [comment]
I would prefer "Direct and diffuse estimation" over "DNI estimation models" and either over "Decomposition models" (novice users may not know what "decomposition" means.) I would put the `dni` function here.

The current "Decomposing and combining irradiance" looks like a mini-junk drawer of functions that are useful but aren't specific models. What about "Transposing irradiance"? and moving `complete_irradiance` to Methods for irradiance calculations"? I don't think `get_extra_radiation` belongs here.

**Reply by AdamRJensen** [reply]
> The current "Decomposing and combining irradiance" looks like a mini-junk drawer of functions that are useful but aren't specific models. What about "Transposing irradiance"? and moving `complete_irradiance` to Methods for irradiance calculations"? I don't think `get_extra_radiation` belongs here.

Naming it "transposing irradiance" makes me think of the [pvlib.irradiance.get_total_irradiance](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.get_total_irradiance.html) function and in general seems too similar to the section "Transposition models".

**Comment by kandersolar** [comment]
A proposed broader reorganization of the `Irradiance` section:

```
Transposition models
    pvlib.irradiance.get_total_irradiance
    pvlib.irradiance.get_sky_diffuse
    pvlib.irradiance.get_ground_diffuse
    pvlib.irradiance.beam_component
    pvlib.irradiance.isotropic
    pvlib.irradiance.perez
    pvlib.irradiance.perez_driesse
    pvlib.irradiance.haydavies
    pvlib.irradiance.klucher
    pvlib.irradiance.reindl
    pvlib.irradiance.king
Reverse transposition models
    pvlib.irradiance.ghi_from_poa_driesse_2023
    pvlib.irradiance.gti_dirint
Decomposition models
    pvlib.irradiance.disc
    pvlib.irradiance.dirint
    pvlib.irradiance.dirindex
    pvlib.irradiance.erbs
    pvlib.irradiance.erbs_driesse
    pvlib.irradiance.orgill_hollands
    pvlib.irradiance.boland
    pvlib.irradiance.campbell_norman
    pvlib.irradiance.louche
Angle of incidence
    pvlib.irradiance.aoi
    pvlib.irradiance.aoi_projection
Clearness index models
    pvlib.irradiance.clearness_index
    pvlib.irradiance.clearness_index_zenith_independent
    pvlib.irradiance.clearsky_index
Other
    pvlib.irradiance.get_extra_radiation
    pvlib.irradiance.dni
    pvlib.irradiance.complete_irradiance
    pvlib.irradiance.poa_components
    pvlib.irradiance.diffuse_par_spitters
    pvlib.albedo.inland_water_dvoracek
Class methods  # although I suggest we remove this subsection; class methods are covered elsewhere
    pvlib.pvsystem.PVSystem.get_irradiance
    pvlib.pvsystem.PVSystem.get_aoi
    pvlib.pvsystem.PVSystem.get_iam
```

**Reply by cwhanse** [reply]
+1 from me. Some nits:

- prefer to see Decomposition listed first, just keeping the usual calculation order decomposition then transposition.
- mild preference to see Albedo as its own heading (it is its own code module).

And looking into the "Other" category, I wonder why pvlib has `poa_components`. Different issue from this reorganization.

**Reply by AdamRJensen** [reply]
I also prefer listing decomposition before transposition.

I suggest renaming "Clearness index models" to "Clearness and clearsky index". Several of these functions aren't models, but just simple divisions with some safe guards.

I'm fine listing the albedo function in "Other" as I don't expect it to get much usage. If future functions are added, we can always make a dedicated "Albedo" section again.

``pvlib.irradiance.poa_components`` adds in-plane irradiance components, which makes me think it should be in transposition.

``pvlib.irradiance.dni```could be argued to be in decomposition as this is likely the modeling step using it, but I'm fine leaving it in other.

Last, I would argue that extraterrestrial irradiance deserves its own section. Even though it's only a single function, it's very frequently used and deserves the promotion of having its own subtitle IMHO.

The reason for first proposing these changes in the first place is based on experience teaching students unfamiliar with pvlib.

Last, we could consider dropping the word "models" from the section titles.

```
Decomposition models
    pvlib.irradiance.disc
    pvlib.irradiance.dirint
    pvlib.irradiance.dirindex
    pvlib.irradiance.erbs
    pvlib.irradiance.erbs_driesse
    pvlib.irradiance.orgill_hollands
    pvlib.irradiance.boland
    pvlib.irradiance.campbell_norman
    pvlib.irradiance.louche
Transposition models
    pvlib.irradiance.get_total_irradiance
    pvlib.irradiance.get_sky_diffuse
    pvlib.irradiance.get_ground_diffuse
    pvlib.irradiance.beam_component
    pvlib.irradiance.poa_components
    pvlib.irradiance.isotropic
    pvlib.irradiance.perez
    pvlib.irradiance.perez_driesse
    pvlib.irradiance.haydavies
    pvlib.irradiance.klucher
    pvlib.irradiance.reindl
    pvlib.irradiance.king
Reverse transposition models
    pvlib.irradiance.ghi_from_poa_driesse_2023
    pvlib.irradiance.gti_dirint
Angle of incidence
    pvlib.irradiance.aoi
    pvlib.irradiance.aoi_projection
Clearness and clearsky index
    pvlib.irradiance.clearness_index
    pvlib.irradiance.clearness_index_zenith_independent
    pvlib.irradiance.clearsky_index
Extraterrestrial radiation
    pvlib.irradiance.get_extra_radiation
Other
    pvlib.irradiance.dni
    pvlib.irradiance.complete_irradiance
    pvlib.irradiance.diffuse_par_spitters
    pvlib.albedo.inland_water_dvoracek
```

**Reply by kandersolar** [reply]
The above rename/reorg was implemented in #2570
